r/src/recordbatch.cpp

```diff
@@ -145,3 +127,64 @@ std::shared_ptr<arrow::RecordBatch> ipc___ReadRecordBatch__InputStream__Schema(
   STOP_IF_NOT_OK(arrow::ipc::ReadRecordBatch(schema, stream.get(), &batch));
   return batch;
 }
+
+Status check_consistent_array_size(const std::vector<std::shared_ptr<arrow::Array>>& arrays, int64_t* num_rows) {
+  *num_rows = arrays[0]->length();
+  for(int64_t i = 1; i<arrays.size(); i++) {
+    if (arrays[i]->length() != *num_rows) {
+      return Status::Invalid("All arrays must have the same length");
+    }
+  }
+  return Status::OK();
+}
+
+std::shared_ptr<arrow::RecordBatch> RecordBatch__from_arrays__knwon_schema(const std::shared_ptr<arrow::Schema>& schema, SEXP lst) {
```

Comment:
We need to get the linting patch ready to go to try to stop the entropy

typo "known"